### PR TITLE
vim-patch:8.0.1361

### LINKED ISSUE #2394: Highlight NONE not working properly
```
colorscheme default
hi TabLineFill NONE
hi link TabLineFill TabLine
```

Screenshot:

bug

### PR DESCRIPTION
**Pull request by navaneethrvce**
https://github.com/vim/vim/commit/97ce419201421f65f4764549ed80307a7ef9c7a6
Problem: Some users don't want to diff with hidden buffers
Solution:   Add the "hiddenoff" item to 'diffopt'. (Alisue, closes #2394)

### CODE REVIEW

**Comment by KillTheMule** [comment]
You should delete the patch file, and [this](https://github.com/neovim/neovim/pull/7841/files#diff-cbdb0e908343d2dbfed180b1e1b141cfR509) isn't needed, since neovim does not use features at compile time, but rather only builds with all features.

**Comment by KillTheMule** [comment]
[This](https://github.com/neovim/neovim/pull/7841/commits/8075dc3ca158b007197b197479635aca4fae009d#diff-cbdb0e908343d2dbfed180b1e1b141cfL509) won't work, you've not only deleted the preprocessor directives, but also the code that's supposed to run.

**Comment by navaneethrvce** [comment]
Hi,
I have just deleted the preprocessor directives and the patch file. Like you asked me too.

**Comment by ckelsel** [comment]
@navaneethrvce fix lint error, then check it by running command : `make lint`
```
src/nvim/buffer.c:509:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
src/nvim/buffer.c:509:  if should always use braces  [readability/braces] [5]
src/nvim/buffer.c:509:  Missing space before ( in if(  [whitespace/parens] [5]
src/nvim/buffer.c:510:  /*-style comment found, it should be replaced with //-style.  /*-style comments are only allowed inside macros.  Note that you should not use /*-style comments to document macros itself, use doxygen-style comments for this.  [readability/old_style_comment] [5]
src/nvim/diff.c:1826:  Missing space after ,  [whitespace/comma] [3]
src/nvim/diff.c:1871:  Should have a space between // and comment  [whitespace/comments] [4]
Total errors found: 6
```

File: src/nvim/diff.c [hunk 1]
```diff
@@ -1864,6 +1868,11 @@ bool diffopt_horizontal(void)
   return (diff_flags & DIFF_HORIZONTAL) != 0;
 }
 
+///Return TRUE if 'diffopt' contains 'hiddenoff'
+bool diffopt_hiddenoff(void)
+{
+    return (diff_flags & DIFF_HIDDEN_OFF) !=0;
```
Comment (justinmk) [inline src/nvim/diff.c]: 2-space indent, please.

File: src/nvim/buffer.c [hunk 2]
```diff
@@ -506,6 +506,9 @@ void close_buffer(win_T *win, buf_T *buf, int action, int abort_if_last)
   if (buf->b_nwindows > 0)
     --buf->b_nwindows;
 
+   if(diffopt_hiddenoff() && !unload_buf && buf->b_nwindows == 0)
+    diff_buf_delete(buf); /* Clear 'diff' for hidden buffer. */
```
Comment (justinmk) [inline src/nvim/buffer.c]: This would not pass `make lint`, please run that locally.

**Comment by justinmk** [comment]
There is an empty file named `Problem:` in this change. Please delete it.

**Comment by janlazo** [comment]
This PR looks abandoned to me so I'm including 8.0.1361 in #8847.

**Pull request closed without merging**